apis/cloud.redhat.com/v1alpha1/clowdjobinvocation_types.go

```diff
@@ -34,17 +34,31 @@ type IqeJobSpec struct {
 	// By default, Clowder will set the image on the ClowdJob to be the
 	// baseImage:name-of-iqe-plugin, but only the tag can be overridden here
 	ImageTag string `json:"imageTag,omitempty"`
+
 	// Indiciates the presence of a selenium container
 	// Note: currently not implemented
 	UI UiSpec `json:"ui,omitempty"`
+
 	// sets the pytest -m args
 	Marker string `json:"marker,omitempty"`
+
 	// sets value for ENV_FOR_DYNACONF
 	DynaconfEnvName string `json:"dynaconfEnvName"`
+
 	// sets pytest -k args
 	Filter string `json:"filter,omitempty"`
+
 	// used when desiring to run `oc debug`on the Job to cause pod to immediately & gracefully exit
 	Debug bool `json:"debug,omitempty"`
+
+	// sets value passed to IQE '--requirements' arg
+	Requirements string `json:"requirements,omitempty"`
```

Comment:
Thinking further about this, and seeing the tests, I'm unsure if this API is sufficient. Though I don't want to burden Clowder with having to do translation, I'm unsure if I like the API for this being to pass in escaped python sets. Passing in just CSV as a string would be OK, but passing in an array of strings would be nicer. As the IQE parameters are "probably" unlikely to change, I feel it could be worth the effort to make the API nice here. At the very least we should document the format that is expected. If the goal is to get more developers to start using IQE tests and CJI is one of the interfaces to do that I think we should take the time to look at this a little more carefully.

Reply:
TBH I don't like this syntax either, it is how the IQE CLI arguments are formed (see https://iqe-docs.apps.ocp4.prod.psi.redhat.com/metadata.html#added-functionality). IMO, they should make this cleaner on the IQE side and just pass in simple comma,separated,strings or something like that. I wanted us to keep it a string on the Clowder side because then we are free to change the format of the arg on the IQE side without impacting CJI's. I think I'll push for nicer syntax over on IQE... thoughts?

Reply:
The thing I'm thinking here is that if we make it a simple string, then it will "change" without warning for the end user if it is ever updated. I know you are saying this gives you the ability to freely change on the IQE side, but that change would doubtlessly introduce breakage on everyone currently using it. Unless it was done in a backwards compatible way. That seems slightly harder to achieve with a string, than it would with keeping our API consistent and translating. Another point on the "changing" it would actually be far better to introduce a _new_ parameter so that backwards compatibility can be kept. Again, if we did the translation, this would allow us to keep our API clean, and internally move over to the new way once it was established, deprecating the old way safely before the options are removed in IQE.

Thoughts?

Reply:
But something to consider is that all these env vars get filtered thru the `iqe_runner.sh` which also has the ability to parse/convert the env vars into whatever format the latest iqe-core expects. That script is released within the iqe-core docker image so they go together "1-to-1". How about we make these fields use type `[]string` and set a CSV env var?

So for example:
```yaml
requirements:
- req1
- req2
- req3
```

Would cause Clowder to set IQE_REQUIREMENTS on the pod to `"req1,req2,req3"` -- and @psegedy will edit `iqe_runner.sh` to parse the CSV env var and run iqe with the argument formatted in the right way (which right now needs to look like `{"req1", "req2", "req3"}` even though I am not a fan of this CLI format and hope it can soon be changed to plain CSV)

If IQE changes the way that argument is formatted, we would only update iqe_runner.sh

Reply:
See https://github.com/RedHatInsights/clowder/pull/452/commits/95e4040b1b927a2f8ea1280ff4954a54c74f0499

Reply:
Well, ok, can't do the 'empty' struct comparison using slices, switched to using pointers to a slice. https://github.com/RedHatInsights/clowder/pull/452/commits/733bfff3b39e3917af83d84b734535d22a650ccf